checker.c

```diff
@@ -1,21 +1,82 @@
-#include <stdio.h>
-#include <assert.h>
-
-int batteryIsOk(float temperature, float soc, float chargeRate) {
-  if(temperature < 0 || temperature > 45) {
-    printf("Temperature out of range!\n");
-    return 0;
-  } else if(soc < 20 || soc > 80) {
-    printf("State of Charge out of range!\n");
-    return 0;
-  } else if(chargeRate > 0.8) {
-    printf("Charge Rate out of range!\n");
-    return 0;
-  }
-  return 1;
-}
-
-int main() {
-  assert(batteryIsOk(25, 70, 0.7));
-  assert(!batteryIsOk(50, 85, 0));
-}
+#include <stdio.h>
+#include <assert.h>
+
+#define NUM_OF_PARAMETERS	3
+
+typedef struct batteryParam_s
+{
+	char name[20];
+	float lowThreshold;
+	float hiThreshold;
+} batteryParam_t;
+
+const batteryParam_t batteryParameters[NUM_OF_PARAMETERS] =
+{
+	{"Temperature", 0.0F, 45.0F},
+	{"State of charge", 20.0F, 80.0F},
+	{"Charge rate", 0.0F, 0.8F}
+};
+
+const float testBatteryReadings[8][NUM_OF_PARAMETERS] =
+{
+	{20.0F, 35.0F, 0.6F},

```

Comment:
The first one is ok, while the rest are not.
Think of making it explicit in the variable names itself - why not have the first one as `batteryReadingsInRange` and the rest of it in a separate array called `batteryReadingsOutOfRange`